fluentbit/conf/conf_test.go

```diff
@@ -530,6 +530,70 @@ func TestSyslogErrors(t *testing.T) {
 	}
 }
 
+func TestWinlog(t *testing.T) {
+	tests := []struct {
+		wineventlog          WindowsEventlog
+		expectedWinlogConfig string
+	}{
+		{
+			wineventlog: WindowsEventlog{
+				Tag:          "windows_event_log",
+				Channels:     "System,Application,Security",
+				Interval_Sec: "1",
+				DB:           "test_DB",
+			},
+			expectedWinlogConfig: `[INPUT]
+    # https://docs.fluentbit.io/manual/pipeline/inputs/windows-event-log
+    Tag            windows_event_log
+    Name           winlog
+    Channels       System,Application,Security
+    Interval_Sec   1
+    DB             test_DB`,
+		},
+	}
+	for _, tc := range tests {
+		got, err := tc.wineventlog.renderConfig()
+		if err != nil {
+			t.Errorf("got error: %v, want no error", err)
+			return
+		}
+		if diff := diff.Diff(tc.expectedWinlogConfig, got); diff != "" {
+			t.Errorf("Tail %v: ran wineventlog.renderConfig() returned unexpected diff (-want +got):\n%s", tc.wineventlog, diff)
+		}
+	}
+}
+
+func TestWinlogErrors(t *testing.T) {
+	tests := []struct {
+		name   string
+		wineventlog WindowsEventlog
+	}{
+		{
+			name: "empty channels",
+			wineventlog: WindowsEventlog{
+				Tag:          "test_tag",
+				Channels:     "",
```

Comment:
Do we error out as expected when `Channels` is not a list of strings separated by `,`? e.g. "System;Applicationn", "System@Application"

Reply:
Channels in winlog plugin or Path in tail plugin both takes comma as separator.  I don't see we have any @ or ; check for the existing tail plugin. If we are going to have such check, we better do it with all fields that only allow comma separated list. I will add a ToDO.